### DISCUSSION TITLE
about call recording

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2377] [Q&A]
There was request to add call recording to my baresip Android app.  Can recording be added/removed to/from an on-going call by loading/unloading sndfile module?

**Comment by juha-h** [comment]
I tried during call give menu command `/insmod sndfile`, but audio dump files were not created.  They were created if `sndfile` was loaded before the call started. So looks like it is not possible to turn on recording while call is on-going.  User interface wise it would have been simpler to have a recording on/off button that is only shown during the call.

**Comment by alfredh** [comment]
when an audio filter module is loaded, it register itself with the global list of audio filters.
it does not update any ongoing calls.

it might be possible to add a new api for `struct call` to add/remove/refresh the list of audio filters.

or keep sndfile.so loaded all the time, and specify list of filters to call:

```c
call_set_audiofilters(call, "webrtc_aec,sndfile,vumeter);
```

**Reply by juha-h** [reply]
Yes, the feature should be call specific, since there may be several simultaneous calls.  The second option may be simpler, since only one new API call is needed.

Regarding order of filters, by default order is this:
```
# Audio filter Modules (in encoding order)
module			auconv.so
module			auresamp.so
#module			vumeter.so
#module			sndfile.so
#module			plc.so
#module			webrtc_aec.so
```
In your answer above `webrtc_aec` is first instead last, which may make more sense if it causes removal of echo from the recording,.

**Comment by alfredh** [comment]
the api belongs to the audio object, and the tx/rx filters should be separate:

example

```c
void audio_set_tx_filters(struct audio *au, const char *filters);
void audio_set_rx_filters(struct audio *au, const char *filters);
```
